AGENTS.md

```diff
@@ -0,0 +1,212 @@
+
+Avoid redundant and verbose comments. Use terse comments and only add comments if it complements, not repeats the code.
+
+Focus on making implementation as small and elegant as possible. Avoid unnecessary loops and allocations. If you see an opportunity of making things simpler by dropping or relaxing some requirements, ask user about the trade-off.
+
+Use modern idiomatic Golang features (version 1.24+). Specifically:
+ - Use for-range for integer iteration where possible. Instead of for i:=0; i < X; i++ {} you must write for i := range X{}.
+ - Use builtin min() and max() where possible (works on any type and any number of values).
+ - Do not capture the for-range variable, since go 1.22 a new copy of the variable is created for each loop iteration.
+
+# Commands
+
+Use "git rm" to remove and "git mv" to rename files instead of directly modifying files on FS.
+
+Do not run “go test ./..." in the root folder as that will start long running integration tests. To test the whole project run "go build && make lint test" in root directory. However, prefer running tests for specific packages instead.
+
+If asked to rebase, always prefix each git command with appropriate settings so that it never launches interactive editor.
+GIT_EDITOR=true GIT_SEQUENCE_EDITOR=true VISUAL=true GIT_PAGER=cat git fetch origin main &&
+GIT_EDITOR=true GIT_SEQUENCE_EDITOR=true VISUAL=true GIT_PAGER=cat git rebase origin/main
+
+# Python
+
+When writing Python scripts, we bias for conciseness. We think of Python in this code base as scripts.
+ - use Python 3.11
+ - Do not catch exceptions to make nicer messages, only catch if you can add critical information
+ - use pathlib.Path in almost all cases over os.path unless it makes code longer
+ - Do not add redundant comments.
+ - Try to keep your code small and the number of abstractions low.
+ - After done, format you code with "ruff format -n <path>"
+ - Use "#!/usr/bin/env python3" shebang.
+
+# Mutators
+
+Mutators should have a structure as follows:
+
+package mutator
+
+type applySomeChange struct{}
+
+// ApplySomeChange applies some change to the bundle.
+func ApplySomeChange() *applySomeChange {
+	return &applySomeChange{}
+}
+
+func (m *applySomeChange) Name() string {
+	return "ApplySomeChange"
+}
+
+func (m *applySomeChange) Apply(ctx context.Context, b *bundle.Bundle) diag.Diagnostics {
+	...
+}
+
+In the code above, notice how the mutator name is used in
+a public function, a struct, and the Name() method.
+The mutator name should start with a verb such as Apply
+and should be used consistently in the code.
+
+# Tests
+
+Each file like process_target_mode_test.go should have a corresponding test file
+like process_target_mode_test.go. If you add new functionality to a file,
+the test file should be extended to cover the new functionality.
+
+Tests should look like the following:
+
+package mutator_test
+
+func TestApplySomeChangeReturnsDiagnostics(t *testing.T) {
```

Comment:
In my experience, all models already know how to write golang tests. What we should mention is to use testify library always, otherwise it starts using stdlib.